third_party/move/tools/move-package/src/compilation/compiled_package.rs

```diff
@@ -675,11 +682,23 @@ impl CompiledPackage {
         };
         let mut root_compiled_units = vec![];
         let mut deps_compiled_units = vec![];
+        let obtain_package_name = |default: Option<Symbol>, source_path_str: &str| -> Symbol {
+            if default.is_none() && source_package_map.contains_key(source_path_str) {
+                *source_package_map.get(source_path_str).unwrap()
```

Comment:
Hmm. Should we have a test for this path?

Reply:
When running several tests in `move-package` such as `tests/test_sources/compilation/one_dep_renamed`, `tests/test_sources/compilation/one_dep_assigned_address` in V2, this path is covered. Since CI does not run V2 by default, it is not shown in CI yet.